cmd/algorand-indexer/daemon.go

```diff
@@ -135,13 +142,24 @@ var daemonCmd = &cobra.Command{
 				_, err := importer.EnsureInitialImport(db, genesisReader, logger)
 				maybeFail(err, "importer.EnsureInitialImport() error")
 				logger.Info("Initializing block import handler.")
+				imp := importer.NewImporter(db)
 
-				nextRound, err := db.GetNextRoundToAccount()
-				maybeFail(err, "failed to get next round, %v", err)
-				bot.SetNextRound(nextRound)
+				logger.Info("Initializing local ledger.")
+				genesisReader = importer.GetGenesisFile(genesisJSONPath, bot.Algod(), logger)
+				genesis, err := readGenesis(genesisReader)
+				maybeFail(err, "Error reading genesis file")
+				genesisBlock, err := getGenesisBlock(bot.Algod())
+				maybeFail(err, "Error getting genesis block")
+				initState, err := util.CreateInitState(&genesis, &genesisBlock)
+				maybeFail(err, "Error creating init state")
 
-				imp := importer.NewImporter(db)
-				handler := blockHandler(imp, 1*time.Second)
+				proc, err := blockprocessor.MakeProcessor(initState, indexerDataDir, imp.ImportBlock)
```

Comment:
Could this constructor work with `MakeProcessor(&genesis, indexerDataDir, imp.ImportBlock)`? It seems like we would want to start fetcher from round 0, so AddBlock could create the initial state when that occurs.

Reply:
how about pass an algod client to `MakeProcessor` and skip passing block 0 to handler in `MakeProcessor`?

Reply:
updated the method to accept a genesis and a genesis block.